src/snappy_device_agents/devices/maas2/maas_storage.py

```diff
@@ -474,9 +482,14 @@ def process_format(self, device):
                 "parent disk block-id": device["parent_disk_blkid"],
             }
         )
-        # format partition
-        if "volume" in device:
+        if device.get("volume"):
```

Comment:
I'm not sure I understand the advantage here to replacing if "volume" in device. Are you trying to validate that it has some sort of good data there that is not going to evaluate to False (such as an empty string, dict, etc?) It might be better to actually validate that either here or in _get_format_partition_id rather than chance it?

Reply:
Yeah, I wanted to add an additional layer of truthiness as you were eluding to (a falsy value like None, 0, "" etc). Also it lines up with the other conditionals which are calling device.get(key).